File: API/Dtos/ReservationReadDto.cs
using System.Collections.Generic;
using API.Infrastructure.Classes;
using API.ViewModels;

namespace API.Dtos {

    public class ReservationReadDto {

        public string ReservationId { get; set; }

        public string Date { get; set; }
        public string RefNo { get; set; }
        public int Adults { get; set; }
        public int Kids { get; set; }
        public int Free { get; set; }
        public int TotalPax { get; set; }
        public string TicketNo { get; set; }
        public string Email { get; set; }
        public string Phones { get; set; }
        public string Remarks { get; set; }
        public string UserId { get; set; }

        public SimpleEntity Customer { get; set; }
        public SimpleEntity Destination { get; set; }
        public PickupPointActiveVM PickupPoint { get; set; }
        public SimpleEntity Driver { get; set; }
        public SimpleEntity Ship { get; set; }

        public List<PassengerReadDto> Passengers { get; set; }

    }

}

File: FrontEnd/src/app/features/reservations/classes/dtos/form/reservation-read-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { PassengerReadDto } from './passenger-read-dto'
import { PickupPointDropdownVM } from '../../../../pickupPoints/classes/view-models/pickupPoint-dropdown-vm'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface ReservationReadDto {

    reservationId: Guid
    customer: SimpleEntity
    destination: SimpleEntity
    driver: SimpleEntity
    pickupPoint: PickupPointDropdownVM
    port: SimpleEntity
    ship: SimpleEntity
    date: string
    refNo: string
    email: string
    phones: string
    remarks: string
    adults: number
    kids: number
    free: number
    totalPax: number
    ticketNo: string
    passengers: PassengerReadDto

}

